src/Tools/SchemaValidator.php

```diff
@@ -329,9 +329,8 @@ private function validatePropertiesTypes(ClassMetadata $class): array
             array_filter(
                 array_map(
                     function (FieldMapping $fieldMapping) use ($class): string|null {
-                        $fieldName = $fieldMapping->fieldName;
-                        assert(isset($class->reflFields[$fieldName]));
-                        $propertyType = $class->reflFields[$fieldName]->getType();
+                        $fieldName    = $fieldMapping->fieldName;
+                        $propertyType = (new ReflectionProperty($fieldMapping->declared ?: $class->name, $fieldName))->getType();
```

Comment:
This may fail with `Embedded` mapping:

```
    #[ORM\Embedded(class: 'EmbeddableDummy')]
    #[Groups(['friends'])]
    public ?EmbeddableDummy $embeddedDummy = null;
```

The `fieldName` will be: `embeddedDummy.dummyName` which leads to a RuntimeException: 

```
Property ApiPlatform\Tests\Fixtures\TestBundle\Entity\RelatedDummy::$embeddedDummy.dummyName does not exist
```

I'm unsure whereas the FieldMapping is wrong or if it should get fixed here.

Reply:
should I open a new issue for that?

Reply:
I have the same issue with `Embedded`.

Reply:
@soyuka @hlecorche can you please test if https://github.com/doctrine/orm/pull/11873 fixes the problem?

Reply:
@beberlei Yes, it works. Thanks